pkg/controllers/multicluster/utils.go

```diff
@@ -134,36 +137,72 @@ func PortsEqualIgnoreOrder(a, b []*model.Port) (equal bool) {
 	return true
 }
 
+func IPsEqualIgnoreOrder(a, b []string) (equal bool) {
+	if len(a) != len(b) {
+		return false
+	}
+
+	aMap := make(map[string]bool)
+	for _, ipA := range a {
+		aMap[ipA] = true
+	}
+
+	for _, ipB := range b {
+		if !aMap[ipB] {
+			return false
+		}
+	}
+	return true
+}
+
+// GetClusterIpsFromServices returns list of ClusterIPs from services
+func GetClusterIpsFromServices(services []*v1.Service) []string {
+	clusterIPs := make([]string, 0)
+	for _, svc := range services {
+		clusterIPs = append(clusterIPs, svc.Spec.ClusterIP)
+	}
+	return clusterIPs
+}
+
 // DerivedName computes the "placeholder" name for an imported service
-func DerivedName(namespace string, name string) string {
+func DerivedName(namespace string, name string, clusterId string) string {
 	hash := sha256.New()
-	hash.Write([]byte(namespace + name))
+	hash.Write([]byte(namespace + name + clusterId))
 	return "imported-" + strings.ToLower(base32.HexEncoding.WithPadding(base32.NoPadding).EncodeToString(hash.Sum(nil)))[:10]
 }
 
 // CreateServiceImportStruct creates struct representation of a ServiceImport
-func CreateServiceImportStruct(namespace string, name string, servicePorts []*model.Port) *multiclusterv1alpha1.ServiceImport {
+func CreateServiceImportStruct(namespace string, name string, clusterIds []string, servicePorts []*model.Port) *multiclusterv1alpha1.ServiceImport {
 	serviceImportPorts := make([]multiclusterv1alpha1.ServicePort, 0)
 	for _, port := range servicePorts {
 		serviceImportPorts = append(serviceImportPorts, PortToServiceImportPort(*port))
 	}
 
+	clusters := make([]multiclusterv1alpha1.ClusterStatus, 0)
+	for _, clusterId := range clusterIds {
+		clusters = append(clusters, multiclusterv1alpha1.ClusterStatus{
+			Cluster: clusterId,
+		})
+	}
+
 	return &multiclusterv1alpha1.ServiceImport{
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace:   namespace,
-			Name:        name,
-			Annotations: map[string]string{DerivedServiceAnnotation: DerivedName(namespace, name)},
```

Comment:
Can we keep `DerivedServiceAnnotation` annotation? The reason being it will help with understanding the state if derived service is removed outside the controller. The value of annotation field is flexible, so we can put json here maybe?
Something like `[{ "cluster-id": "my-cluster1", "derived-service": "imported-xyz"}, { "cluster-id": "my-cluster2", "derived-service": "imported-pqr"}]`

Reply:
@astaticvoid do you have an opinion on this?

Reply:
I agree we should keep the annotation, and the json representation makes sense to me.
Maybe we can be creative with where we create the derived name, possibly after the clusterIdToEndpointsMap.

Reply:
Added back the annotation but will not be using it for anything.
Not worth using the annotation for retrieving the derived service name as it requires unmarshalling the json object every time we want to access and therefore is less efficient that just retrieving the name thru the DerivedName() hash.
Benchmarks:
```
c1 map[cluster-id:test-mcs-clusterid-2 derived-service:imported-us18hivtol]
unmarshalling time: 25.292µs
c2 imported-us18hivtol
taking DerivedService directly time: 6.367µs
```